user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementMetadataUploadDialog.java

```diff
@@ -333,17 +332,22 @@ private void showErrorNotification(String title, String description) {
   }
 
   private void onCanceled() {
-    CancelConfirmationNotificationDialog cancelDialog = new CancelConfirmationNotificationDialog()
-        .withBodyText("Uploaded information has not yet been saved.")
-        .withConfirmText("Discard upload")
-        .withTitle("Discard uploaded information?");
-    cancelDialog.open();
-    cancelDialog.addConfirmListener(event -> {
-      cancelDialog.close();
+
+    NotificationDialog dialog = NotificationDialog.warningDialog()
+        .withTitle("Discard uploaded information?")
+        .withContent(new Span("Uploaded information has not yet been saved."));
+    dialog.setCancelable(true);
+    dialog.setCancelText("Continue Editing");
+    Button redButton = new Button("Discard upload");
+    redButton.addClassName("danger");
+    dialog.setConfirmButton(redButton);
+    dialog.addConfirmListener(event -> {
+      event.getSource().close();
```

Comment:
shouldn't this be taken over by CancelConfirmationDialogFactory ?

Reply:
did it